Tasks/NuGetInstaller/nugetinstaller.ts

```diff
@@ -47,62 +52,110 @@ var runnuget = function(fn) {
+    var filesList: string[];
+    if (solution.indexOf('*') == -1 && solution.indexOf('?') == -1) {
+        // No pattern found, check literal path to a single file
+        tl.checkPath(solution, 'files');
 
-    // Find app files matching the specified pattern
-    tl.debug('Matching glob pattern: ' + solution);
+        // Use the specified single file
+        filesList = [solution];
 
-    // First find the most complete path without any matching patterns
-    var idx = firstWildcardIndex(solution);
-    tl.debug('Index of first wildcard: ' + idx);
-    var findPathRoot = path.dirname(solution.slice(0, idx));
+    } else {
+        var firstWildcardIndex = function (str) {
+            var idx = str.indexOf('*');
 
-    tl.debug('find root dir: ' + findPathRoot);
+            var idxOfWildcard = str.indexOf('?');
+            if (idxOfWildcard > -1) {
+                return (idx > -1) ?
+                    Math.min(idx, idxOfWildcard) : idxOfWildcard;
+            }
 
-    // Now we get a list of all files under this root
-    var allFiles = tl.find(findPathRoot);
+            return idx;
+        }
 
-    // Now matching the pattern against all files
-    filesList = tl.match(allFiles, solution, {matchBase: true});
+        // Find app files matching the specified pattern
+        tl.debug('Matching glob pattern: ' + solution);
 
-    // Fail if no matching .sln files were found
-    if (!filesList || filesList.length == 0) {
-        tl.error('No matching files were found with search pattern: ' + solution);
-        tl.exit(1);
+        // First find the most complete path without any matching patterns
+        var idx = firstWildcardIndex(solution);
+        tl.debug('Index of first wildcard: ' + idx);
+        var findPathRoot = path.dirname(solution.slice(0, idx));
+
+        tl.debug('find root dir: ' + findPathRoot);
+
+        // Now we get a list of all files under this root
+        var allFiles = tl.find(findPathRoot);
+
+        // Now matching the pattern against all files
+        filesList = tl.match(allFiles, solution, { matchBase: true });
+
+        // Fail if no matching .sln files were found
+        if (!filesList || filesList.length == 0) {
+            tl.error('No matching files were found with search pattern: ' + solution);
+            tl.exit(1);
+        }
     }
-}
 
-var result = Q({});
-filesList.forEach((fn) => {
-    result = result.then(() => {
-        return runnuget(fn);
+    var result = Q({});
+    filesList.forEach((fn) => {
+        result = result.then(() => {
+            return runnuget(fn);
+        })
     })
-})
 
-result.then(() => {
-    tl.exit(0);
-})
-.fail((err) => {
-    tl.error(err);
-    tl.exit(1);
-});
+    result.then(() => {
+        removeTempConfig();
+        tl._writeLine("packages are installed successfully.");
+        tl.exit(0);
+    })
+        .fail((err) => {
+            tl.error(err)
+            removeTempConfig();
```

Comment:
Would you like to remove the temp config file in case of failure? Might be better to keep it around though I don't see it getting deleted currently on my Mac.